AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java

```diff
@@ -0,0 +1,175 @@
+package com.ooyala.sample.utils;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.net.wifi.WifiManager;
+import android.os.Handler;
+import android.os.Message;
+import android.view.Gravity;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.ooyala.android.OoyalaException;
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.item.Stream;
+import com.ooyala.android.player.StreamPlayer;
+import com.ooyala.android.util.DebugMode;
+
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+
+
+/**
+ * Created by achaudhari on 9/12/16.
+ */
+public class SampleVideoStreamPlayer extends StreamPlayer {
+
+    private static final String TAG = SampleVideoStreamPlayer.class.getSimpleName();
+
+    private int DURATION;
+    private final int REFRESH_RATE = 250;
+
+    private Timer _timer;
+    private int _playhead = 0;
+    private Handler _timerHandler;
+    private TextView textView;
+    private LinearLayout _linearLayout;
+
+    private Stream stream;
+
+
+    @Override
+    public void init(OoyalaPlayer parent, Set<Stream> streams) {
+        WifiManager wifiManager = (WifiManager)parent.getLayout().getContext().getSystemService(Context.WIFI_SERVICE);
+        boolean isWifiEnabled = wifiManager.isWifiEnabled();
+
+        stream = Stream.bestStream(streams, isWifiEnabled);
+        if (stream == null) {
+            DebugMode.logE(TAG, "ERROR: Invalid Stream (no valid stream available)");
+            this._error = new OoyalaException(OoyalaException.OoyalaErrorCode.ERROR_PLAYBACK_FAILED, "Invalid Stream");
+            setState(OoyalaPlayer.State.ERROR);
+            return;
+        }
+
+        if (parent == null) {
+            this._error = new OoyalaException(OoyalaException.OoyalaErrorCode.ERROR_PLAYBACK_FAILED, "Invalid Parent");
+            setState(OoyalaPlayer.State.ERROR);
+            return;
+        }
+        DURATION = parent.getCurrentItem().getDuration() != 0 ? parent.getCurrentItem().getDuration() : 15000 ;
+        setParent(parent);
+        initPlayer();
+    }
+
+    public void initPlayer () {
+        _linearLayout = new LinearLayout(_parent.getLayout().getContext());
+        _linearLayout.setLayoutParams(new FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT, Gravity.CENTER));
+        _linearLayout.setBackgroundColor(Color.BLACK);
+        textView = new TextView(_parent.getLayout().getContext());
+        _linearLayout.addView(textView);
+        _parent.getLayout().addView(_linearLayout);
```

Comment:
add a comment, that this is where you specifically add your player's view to the parent view